src/item.cpp

```diff
@@ -536,9 +536,7 @@ long item::charges_per_volume( const units::volume &vol ) const
     if( type->volume == 0 ) {
         return INFINITE_CHARGES; // TODO: items should not have 0 volume at all!
     }
-    // Type cast to prevent integer overflow with large volume containers like the cargo dimension
-    return count_by_charges() ? vol * static_cast<int64_t>( type->stack_size ) / type->volume
-           : vol / volume();
+    return count_by_charges() ? vol * type->stack_size / type->volume : vol / volume();
```

Comment:
This change has broken the unit test I wrote.  I feel like it's better to leave this cast, since it's useful for things just temporarily larger than `INT_MAX`.  But if it has to go then you'll need to update the test.

Reply:
Yup, that's reasonable. I have put back the internal typecast.